src/v3/src/transformer/transaction/transform.ts

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (c) 2023-present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import { IdxRemediation } from '@okta/okta-auth-js';
+import { flow } from 'lodash';
+
+import { IDX_STEP } from '../../constants';
+import { TransformStepFnWithOptions } from '../../types';
+
+export const PIV_TYPE = 'X509';
+
+const transformRemediationNameForIdp: TransformStepFnWithOptions = (options) => (formbag) => {
+  const { transaction: { neededToProceed: remediations, nextStep } } = options;
+  if (!remediations.length) {
+    return formbag;
+  }
+
+  // update Remediations array
+  remediations.forEach((remediation: IdxRemediation) => {
+    // TODO: OKTA-504638 Can Add IDP remediation name renaming logic here
+    if (remediation.name === IDX_STEP.REDIRECT_IDP && remediation.type === PIV_TYPE) {
+      // piv idp
+      // eslint-disable-next-line no-param-reassign
```

Comment:
we can in general allow this rule for the whole function using the block-wise eslint-ignore (or even use a .eslintrc in ths transformer/ directory) if it's easier since the entire purpose of these transformers is to mutate the input formBag